### PR TITLE
fix: resolve ts-strict and angular compatibility

### PR DESCRIPTION
**Pull request by miettal**
This change add non-null assertion to angular mandatory component inputs and marked it as `required` in input decolation argument.

problem background

 - typescript strictPropertyInitialization option check non-initialized property.
 - In angular mandatory component inputs has no initialization in ts code.
 - strictPropertyInitialization dont consider it, so the checking point mandatory component inputs as error.

detail procedure for addressing problem

 1. finding non-initialized property in angular components.
 2. regarding these inputs will be initialized in angular lifecycle (=> mandatory component inputs).
 3. marking these inputs peroperty type as non-null assertion type(!).
 3. marking these inputs as required directive({ required: true }).

### CODE REVIEW

File: ui/src/app/shared/genericComponents/flat/flat-widget-line/flat-widget-line.ts [hunk 1]
```diff
@@ -9,10 +8,10 @@ import { AbstractFlatWidgetLine } from "../abstract-flat-widget-line";
...
     /** Name for parameter, displayed on the left side */
-    @Input()
-    public name: string;
+    @Input({ required: true })
+    public name!: string;
 
     /** Width of left Column, right Column is (100 - width of left Column) */
     @Input()
-    public leftColumnWidth: number;
+    public leftColumnWidth?: number;
```
Comment (miettal) [inline ui/src/app/shared/genericComponents/flat/flat-widget-line/flat-widget-line.ts:16]: this input has no initialization so I marked optional.

File: ui/src/app/shared/genericComponents/flat/flat.ts [hunk 2]
```diff
@@ -9,14 +9,14 @@ import { Icon } from 'src/app/shared/type/widget';
 export class FlatWidgetComponent {
 
   /** Image in Header */
-  @Input() public img: string;
+  @Input() public img?: string;
```
Comment (miettal) [inline ui/src/app/shared/genericComponents/flat/flat.ts:12]: this input has no initialization so I marked optional.

File: ui/src/app/shared/genericComponents/chart/abstracthistorychart.ts [hunk 3]
```diff
@@ -37,9 +37,9 @@ export abstract class AbstractHistoryChart implements OnInit {
   @Input() public chartTitle: string = "";
 
   /** TODO: workaround with Observables, to not have to pass the period on Initialisation */
-  @Input() public component: EdgeConfig.Component;
-  @Input() public showPhases: boolean;
-  @Input() public showTotal: boolean;
+  @Input() public component?: EdgeConfig.Component;
```
Comment (miettal) [inline ui/src/app/shared/genericComponents/chart/abstracthistorychart.ts:40]: this input has no initialization so I marked optional.

File: ui/src/app/shared/genericComponents/chart/abstracthistorychart.ts [hunk 4]
```diff
@@ -37,9 +37,9 @@ export abstract class AbstractHistoryChart implements OnInit {
...
+  @Input() public showPhases: boolean = false;
```
Comment (miettal) [inline ui/src/app/shared/genericComponents/chart/abstracthistorychart.ts:41]: this input has no initialization and this is boolean so I add default value false.

File: ui/src/app/shared/genericComponents/chart/chart.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ export class ChartComponent implements OnInit, OnChanges {
   @Input() public isPopoverNeeded: boolean = false;
 
   // Manually trigger ChangeDetection through Inputchange
-  @Input() private period: DefaultTypes.PeriodString;
+  @Input() private period?: DefaultTypes.PeriodString;
```
Comment (miettal) [inline ui/src/app/shared/genericComponents/chart/chart.ts:26]: this input has no initialization so I marked optional.

**Review by sfeilmeier — APPROVED** [review]

**Comment by sfeilmeier** [comment]
Thank you! 🙌

**Pull request merged**